src/ui/SubredditPage.tsx

```diff
@@ -48,38 +55,24 @@ export default class SubredditPage extends Page implements common.SubredditView
   }
 
   get items() {
-    return this._items.concat(); // safe copy
+    return Array.from(this._items).concat(); // safe copy
   }
 
   clear() {
-    this._items = [];
-    this.collectionView.itemCount = 0;
+    this._items = new List();
   }
 
   addItems(newItems: RedditPost[]) {
     this.loading = false;
-    const insertionIndex = this._items.length;
-    this._items = this._items.concat(newItems);
-    this.collectionView.insert(insertionIndex, newItems.length);
-    this.collectionView.refreshIndicator = false;
+    this._items.push(...newItems);
+    this.listView.refreshIndicator = false;
   }
 
-  createCell(type: string) {
-    const cell = isList(type) ? <RedditListCell/> : <RedditGalleryCell/>;
-    cell.onTap(() => {
-      const index = cell.parent(CollectionView).itemIndex(cell);
-      this.onItemSelected.trigger({item: this._items[index]});
-    });
-    return cell;
-  }
-
-  private handleLastVisibleIndexChanged = ({value}: PropertyChangedEvent<CollectionView, number>) => {
-    if (this._items.length - value < (20 / this.collectionView.columnCount) && !this.loading) {
+  private handleLastVisibleIndexChanged = ({value}: PropertyChangedEvent<ListView<RedditPost>, number>) => {
```

Comment:
Since we area only interested in `value` the type could just be `{value: number}`

Reply:
Only because it's shorter